### PR TITLE
Simplify ci yml code

### PR DESCRIPTION
**Pull request by CesarValiente**
@khalp worked on this before her internship finished. 
It adds a nice improvement that removes a lot of duplicated code: iterates through the different projects this repo has and applies them the different `gradle` tasks to build and run our code checks.

### CODE REVIEW

**Comment by joyl1216** [comment]
I also made some changes in this file. Adding a comment title in front of each project to make them more outstanding, such as `# SourceEditor`. We can add the change here, or merge this PR in first, and then I rebased on top of this.

File: .github/workflows/app-samples-CI.yml [hunk 1]
```diff
@@ -5,9 +5,9 @@ name: App samples CI
 
 on:
   push:
-    branches: [ main, hero_notes ]
+    branches: [ main, improve-ci ] # remove improve-ci before merging into main
```
Comment (conceptdev) [inline .github/workflows/app-samples-CI.yml:8]: @CesarValiente we should remove this branch before the merge?

Comment (CesarValiente) [reply .github/workflows/app-samples-CI.yml:8]: yes, correct, goof catch!

**Comment by CesarValiente** [comment]
@conceptdev comments addressed ;-)

**Pull request merged**